torchtune/models/common.py

```diff
@@ -0,0 +1,33 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import torch.nn as nn
+from torchtune.config._utils import _get_component_from_path
+from typing import Dict, Any
+
+from torchtune.modules.tied_linear import TiedLinear
+
+def classifier_model(num_classes: int, base_model: str, **base_model_kwargs: Dict[str, Any]) -> nn.Module:
+    """
+    Create a classifier model from a base model by adapting the output layer.
+
+    Args:
+        num_classes (int): The number of classes for the classifier.
+        base_model (str): The path to the base model.
+        base_model_kwargs (Dict[str, Any]): Keyword arguments for the base model.
+
+    Returns:
+        nn.Module: The classifier model.
+    """
+    model = _get_component_from_path(base_model)(**base_model_kwargs)
```

Comment:
if this is bound by the models in our library, then we would need to test that we can successfully create a classifier model for every model in our library, and any future models that are added. Can users pass in a custom model path? If that's not the intention, then we'll need to add some more gating. For example, if we check for if it's a TransformerDecoder instance that would easily keep this constrained to models that we know with confidence can be made into classifiers.

Reply:
Yep agreed, if we test against TransformerDecoder that should satisfy all the conditions I mentioned